### PR TITLE
mon/OSDMonitor: add check for crush rule size in pool set size command

### PR DESCRIPTION
**Pull request by vumrao**
mon/OSDMonitor: add check for crush rule size in pool set size command
Fixes: https://tracker.ceph.com/issues/42111

Signed-off-by: Vikhyat Umrao <vumrao@redhat.com>

### CODE REVIEW

**Comment by vumrao** [comment]
- Tested in vstart cluster:

~~~
# rules
rule replicated_rule {
        id 0
        type replicated
        min_size 1
        max_size 2
        step take default
        step choose firstn 0 type osd
        step emit
}

# bin/ceph-mon -v
ceph version v15.0.0-5750-gdc473ec733 (dc473ec73336ca4c55bcac5c43496d63ca43f0d1) octopus (dev)

# bin/ceph osd pool set testpool size 3
Error EINVAL: pool size is bigger than the crush rule max size
~~~

**Comment by jdurgin** [comment]
it'd be nice to add a test for this, e.g. in qa/workunits/mon/pool_ops.sh

**Comment by vumrao** [comment]
> it'd be nice to add a test for this, e.g. in qa/workunits/mon/pool_ops.sh

Thanks, Josh for the feedback. I have written a test case, working on testing it locally. I will send a new commit for this test.

**Comment by vumrao** [comment]
@jdurgin Hi Josh - I have done local testing for this test case in my vstart cluster and it works fine. Let me know what do you think.

**Comment by vumrao** [comment]
- The output from the local run in the vstart cluster - 

~~~
+ bin/ceph osd pool set foo size 20
Error EINVAL: pool size must be between 1 and 10
+ return 0
+ bin/ceph osd pool set foo size 3
set pool 3 size to 3
+ bin/ceph osd getcrushmap -o crush
7
+ bin/crushtool -d crush -o crush.txt
+ sed -i 's/max_size 10/max_size 3/' crush.txt
+ bin/crushtool -c crush.txt -o crush.new
+ bin/ceph osd setcrushmap -i crush.new
8
+ expect_false bin/ceph osd pool set foo size 4
+ set -x
+ bin/ceph osd pool set foo size 4
Error EINVAL: pool size is bigger than the crush rule max size
+ return 0
+ bin/ceph osd setcrushmap -i crush
9
~~~

File: qa/workunits/mon/pool_ops.sh [hunk 1]
```diff
@@ -25,6 +25,16 @@ ceph osd pool set foo size 10
 expect_false ceph osd pool set foo size 0
 expect_false ceph osd pool set foo size 20
 
+ceph osd pool set foo size 3
+ceph osd getcrushmap -o crush
+crushtool -d crush -o crush.txt
+sed -i 's/max_size 10/max_size 3/' crush.txt
+crushtool -c crush.txt -o crush.new
+ceph osd setcrushmap -i crush.new
+expect_false bin/ceph osd pool set foo size 4
```
Comment (vumrao) [inline qa/workunits/mon/pool_ops.sh]: Fixing bin/ceph to ceph. It came from my vstart cluster testing.

Comment (vumrao) [reply qa/workunits/mon/pool_ops.sh]: Fixed.

**Comment by vumrao** [comment]
retest this please.

**Comment by tchaikov** [comment]
jenkins test make check

**Pull request merged**